Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -905,20 +907,13 @@ WarpXOpenPMDPlot::SaveRealProperty (ParticleIter& pti,
                        amrex::Vector<std::string> const& int_comp_names) const
 
 {
-  int numOutputReal = 0;
-  int const totalRealAttrs = m_NumAoSRealAttributes + m_NumSoARealAttributes;
-
-  for( int i = 0; i < totalRealAttrs; ++i )
-    if( write_real_comp[i] )
-      ++numOutputReal;
-
   auto const numParticleOnTile = pti.numParticles();
   uint64_t const numParticleOnTile64 = static_cast<uint64_t>( numParticleOnTile );
   auto const& aos = pti.GetArrayOfStructs();  // size =  numParticlesOnTile
   auto const& soa = pti.GetStructOfArrays();
   // first we concatinate the AoS into contiguous arrays
   {
-    for( auto idx=0; idx<m_NumAoSRealAttributes; idx++ ) {
+    for( auto idx=0; idx<ParticleIter::ContainerType::NStructReal; idx++ ) {
```

Comment:
LGTM is smart here, since we don't have extra real attributes yet
```suggestion
    for( auto idx=0; idx<ParticleIter::ContainerType::NStructReal; idx++ ) {  // lgtm [cpp/constant-comparison]
```